python/lsst/obs/decam/decamCpIsr.py

```diff
@@ -0,0 +1,145 @@
+from .isr import DecamIsrTask
+
+
+def _computeEdgeSize(rawExposure, calibExposure):
+    """Compute the number of edge trim pixels of the calibration product.
+    Some Community Pipeline Calibration products are trimmed on their edges
+    and are smaller than the raw data. Use the dimension difference between
+    raw exposure and the calibration product to compute the edge trim pixels.
+    @param[in] rawExposure: the data section of a raw exposure
+    @param[in] calibExposure: calibration bias or flat exposure,
+                              known to be smaller than raw data
+    @return an integer as the number of trimmed pixels on each edge
+    """
+    nx, ny = rawExposure.getBBox().getDimensions() - calibExposure.getBBox().getDimensions()
+    assert nx == ny, "Exposure is trimmed differently in X and Y"
+    assert nx%2 == 0, "Exposure is trimmed unevenly in X"
+    assert nx >= 0, "Calibration image is larger than raw data"
+    return nx/2
+
+
+class DecamCpIsrTask(DecamIsrTask):
+    """Perform ISR task using Community Pipeline Calibration Products MasterCal
+
+    The CP MasterCal products have butler dataset types cpBias and cpFlat,
+    different from the LSST-generated calibration products (bias/flat).
+    """
+    def readIsrData(self, dataRef, rawExposure):
+        """Retrieve necessary frames for instrument signature removal
+        \param[in] dataRef -- a daf.persistence.butlerSubset.ButlerDataRef
+                              of the detector data to be processed
+        \param[in] rawExposure -- a reference raw exposure that will later be
+                                  corrected with the retrieved calibration data;
+                                  should not be modified in this method.
+        \return a pipeBase.Struct with fields containing kwargs expected by run()
+         - bias: exposure of bias frame
+         - dark: exposure of dark frame
+         - flat: exposure of flat field
+         - defects: list of detects
+         - fringeStruct: a pipeBase.Struct with field fringes containing
+                         exposure of fringe frame or list of fringe exposure
+        """
+        ccd = rawExposure.getDetector()
+
+        biasExposure = self.getIsrExposure(dataRef, "cpBias") if self.config.doBias else None
+        # immediate=True required for functors and linearizers are functors; see ticket DM-6515
+        linearizer = dataRef.get("linearizer", immediate=True) if self.doLinearize(ccd) else None
+        darkExposure = self.getIsrExposure(dataRef, "dark") if self.config.doDark else None
+        flatExposure = self.getIsrExposure(dataRef, "cpFlat") if self.config.doFlat else None
+        brighterFatterKernel = dataRef.get("brighterFatterKernel") if self.config.doBrighterFatter else None
+
+        defectList = dataRef.get("defects")
+
+        if self.config.doFringe and self.fringe.checkFilter(rawExposure):
+            self.log.error("Fringe correction with CP products is not verified")
```

Comment:
`self.log` has no attribute `error`.

I'm not actually sure what behaviour you're aiming for here. Is the idea to terminate execution at this point, or to log a warning and continue? If it's the former, you'll need to do something more than log a message; if it's the latter, you'll still need to provide some sort of default `fringeStruct` since you refer to it at line 88, below.